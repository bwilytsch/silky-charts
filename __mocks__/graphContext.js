/* ignore file coverage */

const rect = { top: 10, right: 10, bottom: 10, left: 10 };

module.exports = {
  margin: rect,
  node: { getBoundingClientRect: jest.fn(() => rect) },
  staticTooltip: false,
  visibleTicks: true,
  xAxisLabelRotation: false,
  xAxisLabelRotationValue: -65,
};
