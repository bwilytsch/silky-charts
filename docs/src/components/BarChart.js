import React from 'react';
import styled from 'styled-components';
import { Bar } from 'silky-charts';
import Chart from '../containers/Chart';
import BarData from '../markdown/bar-data.md';
import BarProps from '../markdown/bar-props.md';
import { InlineCode, Pre } from '../ui/code';
import { Table, Td } from '../ui/table';

const data = [
  { name: 'E', value: 0.12702 },
  { name: 'T', value: 0.09056 },
  { name: 'A', value: 0.08167 },
  { name: 'O', value: 0.07507 },
  { name: 'I', value: 0.06966 },
  { name: 'N', value: 0.06749 },
  { name: 'S', value: 0.06327 },
  { name: 'H', value: 0.06094 },
  { name: 'R', value: 0.05987 },
  { name: 'D', value: 0.04253 },
  { name: 'L', value: 0.04025 },
  { name: 'C', value: 0.02782 },
  { name: 'U', value: 0.02758 },
  { name: 'M', value: 0.02406 },
  { name: 'W', value: 0.0236 },
  { name: 'F', value: 0.02288 },
  { name: 'G', value: 0.02015 },
  { name: 'Y', value: 0.01974 },
  { name: 'P', value: 0.01929 },
  { name: 'B', value: 0.01492 },
  { name: 'V', value: 0.00978 },
  { name: 'K', value: 0.00772 },
  { name: 'J', value: 0.00153 },
  { name: 'X', value: 0.0015 },
  { name: 'Q', value: 0.00095 },
  { name: 'Z', value: 0.00074 },
];

const H3 = styled.h3`
  margin-top: 40px;
`;

const BarChart = () => (
  <>
    <h2>Bar Chart</h2>
    <Chart>
      <Bar data={data} responsive tooltip />
    </Chart>
    <H3>Data Definition</H3>
    <BarData components={{ pre: Pre }} />
    <H3>Props</H3>
    <BarProps components={{ inlineCode: InlineCode, table: Table, td: Td }} />
  </>
);

export default BarChart;
