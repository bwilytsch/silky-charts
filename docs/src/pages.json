[
  {
    "title": "Getting Started",
    "path": "/",
    "component": "Introduction"
  },
  {
    "title": "Bar",
    "path": "bar",
    "component": "BarChart"
  }
]
