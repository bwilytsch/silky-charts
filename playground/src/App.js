/* eslint-disable */

import React from 'react';
import styled from 'styled-components';
import { Bar, BarLine, StackedArea, toSeries } from 'silky-charts';
import { aapl, fruits, letters, transport } from './data/';
import Container from './components/Container';
import ChartContainer from './components/ChartContainer';
import 'normalize.css';
import './styles.css';

const App = () => (
  <Container>
    <ChartContainer>
      <StackedArea
        // aspectRatio="4:3"
        data={fruits}
        // dataSource="Unknown"
        // dateFormat="%B %d, %Y"
        // grid
        // height="300"
        // lineSeries={['Sold', 'Lost']}
        // lineType="curveCardinalOpen"
        // lineTypeOption={0.1}
        margin={{ top: 100, right: 100, bottom: 100, left: 100 }}
        // onClick={() => {}}
        // onMouseEnter={() => {}}
        // onMouseLeave={() => {}}
        // padding={1}
        responsive
        // secondaryTheme="madang"
        // stackedSeries={['Pear', 'Mango', 'Papaya']}
        // staticTooltip="top-left"
        // theme="curiousBlue"
        // title="Placeholder Chart Title"
        tooltip
        // width={500}
        // xAxisChartLabel="X axis label"
        // xAxisLabelRotation
        // xAxisLabelRotationValue={-65}
        // xAxisTicks={10}
        // yAxisChartLabel="Y axis label"
        // yAxisTicks={10}
      />
    </ChartContainer>
  </Container>
);

export default App;
