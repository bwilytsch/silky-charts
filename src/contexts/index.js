export { default as GraphContext } from './graph';
