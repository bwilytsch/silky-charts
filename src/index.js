export { default as Bar } from './Bar';
export { default as BarLine } from './BarLine';
export { default as StackedArea } from './StackedArea';

export { default as toSeries } from './utils/toSeries';
