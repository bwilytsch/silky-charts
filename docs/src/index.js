import React from 'react';
import { render } from 'react-dom';
import { Router, Redirect } from '@reach/router';
import Layout from './containers/Layout';
import Header from './containers/Header';
import Navigation from './containers/Navigation';
import Content from './containers/Content';
import Introduction from './components/Introduction';
import BarChart from './components/BarChart';
import pages from './pages.json';
import 'normalize.css';
import './styles.css';

const App = () => {
  const baseURL = '/silky-charts';
  const pathname = window.location.pathname;
  const currentPath = pathname.includes(baseURL)
    ? pathname.replace(baseURL, '')
    : '/';
  const isValidPathname =
    currentPath && pages.map(({ path }) => path).includes(currentPath);

  console.log(currentPath);

  const contentComponents = {
    Introduction,
    BarChart,
  };

  return (
    <Layout>
      <Header />
      <Navigation
        items={pages}
        baseURL={baseURL}
        initial={isValidPathname ? currentPath : '/'}
      />
      <Router primary={false}>
        {currentPath === '/' && (
          <Redirect from={currentPath} to={baseURL} noThrow />
        )}
        <Content path={baseURL}>
          {pages.map(({ component, path }, idx) => {
            const Component = contentComponents[component];
            return <Component key={idx} path={path} default={path === '/'} />;
          })}
        </Content>
      </Router>
    </Layout>
  );
};

render(<App />, document.getElementById('app'));
