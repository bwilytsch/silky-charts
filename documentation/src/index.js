import React from 'react';
import { render } from 'react-dom';
import { Router, Redirect } from '@reach/router';
import Layout from './containers/Layout';
import Header from './containers/Header';
import Navigation from './containers/Navigation';
import Content from './containers/Content';
import Introduction from './components/Introduction';
import BarChart from './components/BarChart';
import pages from './pages.json';
import 'normalize.css';
import './styles.css';

const App = () => {
  const pathname = window.location.pathname;
  const defaultPath = '/';
  const currentPath =
    pathname !== defaultPath ? pathname.replace(defaultPath, '') : pathname;
  const isInvalidPath = !pages
    .map(({ pathname }) => pathname)
    .includes(currentPath);

  const contentComponents = {
    Introduction,
    BarChart,
  };

  return (
    <Layout>
      <Header />
      <Navigation
        items={pages}
        initialPath={isInvalidPath ? defaultPath : currentPath}
      />
      <Content>
        <Router>
          {isInvalidPath && (
            <Redirect from={currentPath} to={defaultPath} noThrow />
          )}
          {pages.map(({ component, pathname }, idx) => {
            const Component = contentComponents[component];
            return (
              <Component
                key={idx}
                path={pathname}
                default={pathname === defaultPath}
              />
            );
          })}
        </Router>
      </Content>
    </Layout>
  );
};

render(<App />, document.getElementById('app'));
