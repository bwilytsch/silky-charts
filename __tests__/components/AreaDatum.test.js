import React from 'react';
import { render, fireEvent, cleanup } from 'react-testing-library';
import { GraphContext } from '../../src/contexts';
import graphContext from '../../__mocks__/graphContext';
import { AreaDatum } from '../../src/components';
import 'jest-styled-components';

jest.mock('../../src/utils/getMousePosition', () => jest.fn(() => [0, 0]));
jest.mock('../../src/utils/getNearestPoint', () => jest.fn(() => 1));

const svg = document.createElement('svg');

const props = {
  area: jest.fn(),
  axis: 'foo',
  dataPositions: [0, 2, 4, 6],
  datum: [{ name: '09/24/1980', series: 'bar', value: 0 }],
  margin: { left: 0 },
  position: { x: 10, y: 10 },
  series: 'bar',
  svg,
  theme: 'monteCarlo',
  tooltipData: {
    0: [{ name: '09/24/1980', series: 'bar', value: 1 }],
    1: [{ name: '09/25/1980', series: 'baz', value: 2 }],
  },
};

afterEach(cleanup);

test('AreaDatum', () => {
  const { container } = render(
    <GraphContext.Provider value={graphContext}>
      <svg>
        <AreaDatum {...props} />
      </svg>
    </GraphContext.Provider>
  );

  const area = container.querySelector('g');
  expect(area).toMatchSnapshot();
  expect(area.classList.contains('bar-layer')).toBe(true);
});

test('AreaDatum:onMouseEnter show tooltip', () => {
  const handleMouseEnter = jest.fn();

  const { container } = render(
    <GraphContext.Provider value={graphContext}>
      <svg>
        <AreaDatum {...props} onMouseEnter={handleMouseEnter} tooltip />
      </svg>
    </GraphContext.Provider>
  );

  const path = container.querySelector('path');
  fireEvent.mouseEnter(path, {});

  expect(handleMouseEnter).toHaveBeenCalled();

  const tooltip = document.querySelector('.silky-charts-tooltip');
  expect(tooltip.nodeName).toBe('DIV');
});

test('AreaDatum:onMouseLeave hide tooltip', () => {
  const handleMouseLeave = jest.fn();

  const { container } = render(
    <GraphContext.Provider value={graphContext}>
      <svg>
        <AreaDatum {...props} onMouseLeave={handleMouseLeave} tooltip />
      </svg>
    </GraphContext.Provider>
  );

  const svg = container.querySelector('svg');
  expect(svg).toMatchSnapshot();

  const path = container.querySelector('path');
  fireEvent.mouseLeave(path, {});

  expect(handleMouseLeave).toHaveBeenCalled();

  const tooltip = document.querySelector('.silky-charts-tooltip');
  expect(tooltip).toBe(null);
});

test('AreaDatum:onMouseMove', () => {
  const { container } = render(
    <GraphContext.Provider value={graphContext}>
      <svg>
        <AreaDatum {...props} tooltip />
      </svg>
    </GraphContext.Provider>
  );

  const path = container.querySelector('path');
  fireEvent.mouseMove(path, { clientX: 10, clientY: 10 });
});
