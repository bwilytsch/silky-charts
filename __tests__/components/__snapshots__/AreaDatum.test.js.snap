// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AreaDatum 1`] = `
.c0 {
  fill: none;
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c0:hover {
  fill: rgb(0,0,0);
}

<g
  class="bar-layer"
>
  <path
    class="line-path stacked-area c0"
    stroke-width="0"
  />
</g>
`;

exports[`AreaDatum:onMouseLeave hide tooltip 1`] = `
.c0 {
  fill: none;
  stroke: none;
  stroke-width: 3;
}

pointer-events: .c0:hover {
  fill: rgb(0,0,0);
}

<svg>
  <g
    class="bar-layer"
  >
    <path
      class="line-path stacked-area c0"
      stroke-width="0"
    />
  </g>
</svg>
`;
